python_modules/dagster/dagster/_config/pythonic_config/__init__.py

```diff
@@ -706,6 +713,13 @@ def initialize_and_run(self, context: InitResourceContext) -> TResValue:
     def _create_object_fn(self, context: InitResourceContext) -> TResValue:
         return self.create_resource(context)
 
+    def get_context(self) -> InitResourceContext:
```

Comment:
So one point of potential confusion here is that there will be a get_context() method in contexts where an OpExecutionContext is also available

e.g.

```
def an_asset(context: OpExecutionContext, a_resource: AResource):
   a_resource.get_context() != context
```

Maybe we should call it `get_resource_context` or `get_context_for_resource`

Reply:
There are definitely problems with those names too

Reply:
are there cases when you would want/need to use the resource's context in an asset/op/etc? If not, then it could just be a private `_get_context` (or `_get_resource_context` etc) method right?

Reply:
I don't think that users will actually call `get_context` in the body of the op. It's more about the conceptual confusion, which I still think would exist with a private `_get_context` method. Meaning that if I type that code, which context am I getting? Is it the same as the one passed to the op? It's already confusion and this risks making it more confusing.

Reply:
cool, that all makes sense to me!

Reply:
`get_resource_init_context` potentially confusing as it is no longer access in an initializing context. It is accessed during execution